### PR TITLE
Apr 2383 sector most experienced employee extra details

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.InternalApi.UnitTests/Services/Assessor/AssessorSectorDetailsServiceTests.cs [hunk 1]
```diff
@@ -116,7 +116,7 @@ public async Task Get_sector_details_for_page3_answers()
         [Test]
         public async Task Get_sector_details_for_page4_answers()
         {
-            _sectorQuestionIds.FullName = "FullNameQuestionId";
+            _sectorQuestionIds.FirstName = "FullNameQuestionId";
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.InternalApi.UnitTests/Services/Assessor/AssessorSectorDetailsServiceTests.cs:119]: Missing Lastname and the value isn't quite right

File: src/SFA.DAS.ApplyService.InternalApi.UnitTests/Services/Assessor/AssessorSectorDetailsServiceTests.cs [hunk 2]
```diff
@@ -127,7 +127,7 @@ public async Task Get_sector_details_for_page4_answers()
             var organisationDetails = "other organisations";
         
             _extractAnswerValueService
-                .Setup(a => a.ExtractAnswerValueFromQuestionId(It.IsAny<List<AssessorAnswer>>(), _sectorQuestionIds.FullName))
+                .Setup(a => a.ExtractAnswerValueFromQuestionId(It.IsAny<List<AssessorAnswer>>(), _sectorQuestionIds.FirstName))
                 .Returns(fullName);
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.InternalApi.UnitTests/Services/Assessor/AssessorSectorDetailsServiceTests.cs:131]: first name. Also do you need last name here?

File: src/SFA.DAS.ApplyService.InternalApi.UnitTests/Services/Assessor/AssessorSectorDetailsServiceTests.cs [hunk 3]
```diff
@@ -143,7 +143,7 @@ public async Task Get_sector_details_for_page4_answers()
                 .Returns(organisationDetails);
             
             var actualSectorDetails = await _service.GetSectorDetails(_applicationId, _firstPageId);
-            Assert.AreEqual(fullName, actualSectorDetails.FullName);
+            Assert.AreEqual(fullName, actualSectorDetails.FirstName);
             Assert.AreEqual(jobRole, actualSectorDetails.JobRole);
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.InternalApi.UnitTests/Services/Assessor/AssessorSectorDetailsServiceTests.cs:147]: missing last name

File: src/SFA.DAS.ApplyService.Web/Services/QuestionPropertyTokeniser.cs [hunk 4]
```diff
@@ -23,7 +23,9 @@ public async Task<string> GetTokenisedValue(Guid applicationId, string tokenised
                 return string.Empty;
             }
 
-            if (ContainsTokens(tokenisedValue))
+            int maxCounter = 5;
+            int startCounter = 0;
+            while (ContainsTokens(tokenisedValue) && startCounter < maxCounter)
```
Comment (martin-lush) [inline src/SFA.DAS.ApplyService.Web/Services/QuestionPropertyTokeniser.cs:28]: I think there's a better way of doing this but it'll do for now. Maybe raise as tech debt

File: src/SFA.DAS.ApplyService.InternalApi.UnitTests/Services/Assessor/AssessorSectorDetailsServiceTests.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.InternalApi.UnitTests/Services/Assessor/AssessorSectorDetailsServiceTests.cs:131]: yes

File: src/SFA.DAS.ApplyService.Web/Services/QuestionPropertyTokeniser.cs (hunk 4 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.Web/Services/QuestionPropertyTokeniser.cs:28]: maybe, but I'm wary of a while loop that assesses a value altered within it, without some maximum counter.  That's how entire apps crash.....

**Review by martin-lush — APPROVED** [review]

**Review by martin-lush — APPROVED** [review]

**Pull request merged**